apps/bot_manager/lib/game_socket_handler.ex

```diff
@@ -46,9 +46,6 @@ defmodule BotManager.GameSocketHandler do
       %{event: {:joined, joined}} ->
         {:ok, Map.merge(state, joined)}
 
-      %{event: {:finished, _}} ->
```

Comment:
Why do you remove this? If this is correct you should delete the terminate/1 callback aswell.

Reply:
> Now the bot socket will be closed when the game closes after 10 seconds of the game being finished

This ws closes when the game finishes and you send a message to its dead process.
That's because we don't close the websockets from the server (we have an issue for that), so you should close the client ws gracefully or you could trigger several errors on Arena.
Try to restore this and stop the spawning of the bots from Arena.